pkg/model/bootstrapscript_test.go

```diff
@@ -57,20 +58,53 @@ func TestBootstrapUserData(t *testing.T) {
 	cs := []struct {
 		Role             kops.InstanceGroupRole
 		ExpectedFilePath string
+		HookSpecRoles    []kops.InstanceGroupRole
 	}{
 		{
 			Role:             "Master",
 			ExpectedFilePath: "tests/data/bootstrapscript_0.txt",
```

Comment:
No worries, I was feeling it might be a bit excessive. :D Each of the test cases have some subtle differences in the instance group type / configuration, which can cause a change in output of the bootstrap script. I covered each where I felt the logic could go a bit wrong, such as multiple hook specs applying to specific / all instance groups. The test file contents all vary towards the bottom, in the generation of the `cluster_spec.yaml` and `ig-spec.yaml` files.